### PR TITLE
Add fsclone integration test & format improvements

### PR DESCRIPTION
**Pull request by mpacer**
This adds integration tests for file system cloning and is based on #162 and #164.

It also required addressing some minor inconsistencies I was discovering when I attempted to try our integration tests with a genuinely empty notebook.

These have to do with how the notebook stores cell `source` values in memory vs. on disk. Because we were using `json.dumps()` to get our to-disk content, we were not always writing out the appropriate information that was being passed through. This was most obvious in the case of an empty source `[]` which was being written to s3 as `''`. 
Using `nbformat.writes` and `nbformat.from_dict` alleviates the issue.

### CODE REVIEW

File: bookstore/clone.py [hunk 1]
```diff
@@ -270,7 +270,7 @@ def build_content_model(self, content, target_path):
         dict
             Jupyter Contents API compatible model
         """
-        path = self.contents_manager.increment_filename(target_path)
+        path = self.contents_manager.increment_filename(target_path, insert='-')
```
Comment (chyzzqo2) [inline bookstore/clone.py]: Seems unrelated, what was the reason for this?

File: bookstore/clone.py [hunk 2]
```diff
@@ -406,3 +406,101 @@ def construct_template_params(self, relpath, fs_clonepath):
...
+
+        The response payload should match the standard Jupyter contents
+        API POST response.
+        """
+        model = self.get_json_body()
+
+        relpath = model.get("relpath", "")
+        fs_clonepath = validate_relpath(relpath, self.bookstore_settings, self.log)
+
+        target_path = model.get("target_path", "") or os.path.basename(os.path.relpath(relpath))
```
Comment (chyzzqo2) [inline bookstore/clone.py]: If we are going to except a target path, we also need to make sure it is inside of the servers notebook directory. Or is that validation already done somewhere else?

File: bookstore/clone.py (hunk 1 above)
Comment (mpacer) [reply bookstore/clone.py]: To create consistency between the S3 and FS clone APIs file name conflict policies. I can yank this out and make a separate PR if you'd like.

Comment (chyzzqo2) [reply bookstore/clone.py]: No it's fine to leave in

File: bookstore/clone.py (hunk 2 above)
Comment (mpacer) [reply bookstore/clone.py]: `self.contents_manager.save(content_model, content_model['path'])` has functionality that strips any prefixed `/` (making it effectively a relpath) and then requests to save it relative to the notebook's base directory.

That in turn runs `_get_os_path` on from the `filio.FileManagerMixin` to make sure that once it's resolved to the real path it is still kept underneath the basedir https://github.com/jupyter/notebook/blob/e498de6775b3de57f5ff827e562c179b17893064/notebook/services/contents/fileio.py#L243-L264
